### ISSUE TITLE
often unable to download firmware on wifi due to "connection time out"  

### ISSUE THREAD

**Issue by ukeconsulting** [fhorinek/BB#49]
firmware 133 until 139

rarely it works, mostly it doesn't work. 

when it happen, after when i retry i have the error prompt directly, i have to reboot the device to reinitiate the download. 
on one device from 138 to 139 i had to put the firmware manually because it never finish downloading the file.

of course the wifi is working and stable in the same time on other computers.

**Comment by ukeconsulting** [comment]
update from 139 to 140: 
one device the one able to be updated by wifi : no problem 
one device the one only once able to be updated by wifi (once for 133 to 138 after a crazy lot of try) : after 2 "connection time out" and so 2 reboot, i am able to download it fully, but when i press the middle button (green check) the prompt box "start update process" disapear and nothing happen. i could redownload it 2 more times (strangely without any reboot) and again the prompt box just disapear when i push the middle button.

**Comment by ukeconsulting** [comment]
i tried a reset and another network : same issue.

**Comment by fhorinek** [comment]
yes I managed to observe this issue few times, however the random nature is making it difficult to catch. Can you please enable debug to file option and send me the debug.log from the failed attempt?

**Comment by ukeconsulting** [comment]
i just made new tests before i read your answer:
 
still same problems, if i check the "install update" menu after download, i didn't see the D0000140.FW file
but this time i tester the "cancel" option after download and when i check the "install update" i could see the D0000140.FW file and i could successfully update manually from the updated file. 

2 remarks : 
- the FW files are not listed in alphabetical order (133, 138,140, 139), if numerous that can be not easy to read and find the good one.
-  i have already 4 FW, there must be a protection to activate to not saturate the memory. example: limit the number of FW and automatically delete the oldest one and always keep the 2 last working one for security.  or at least, be able to delete manually the files. perhaps a file manager could be useful.  


now i read your answer, i reinstalled the 139 and put the debug to file option and tried to update by wifi.
the update worked immediatly oO
so i was again in 140, tried to update to 139, when i say yes nothing happened. same to 138 and 133... but i could update from 140 to 140. 
retry, to 139 NO OK but to 138 OK. 
on 138, the firmware 140 could be downloaded despite i already have the file on the device, and at the "start update process" prompt the yes make the menu disappear and when i check the "install update" menu i have no more D0000140.FW file. 
update test again: connection error, reboot and miracle it updated to 140 after "start update process".
everything was with the debug mode. 
https://transfert.free.fr/download/LRYbJEPYnu4zbkrf5wIICvNgaWE 

note: the battery was full yesterday evening and now it is 61%, i know battery optimisation are not a priority but there will be a issue soon.

**Comment by ukeconsulting** [comment]
i just made new tests before i read your answer:


without the debug mode.
------------------------ 
still same problems, if i check the "install update" menu after download, i didn't see the D0000140.FW file
but this time i tested the "cancel" option after download and when i check the "install update" i could see the D0000140.FW file and i could successfully update manually to 140 from the updated file. 

2 remarks : 
- the FW files are not listed in alphabetical order (133, 138,140, 139), if numerous that can be not easy to read and find the good one.
-  i have already 4 FW, there must be a protection to activate to not saturate the memory. example: limit the number of FW and automatically delete the oldest one and always keep the 2 last working one for security.  or at least, be able to delete manually the files. perhaps a file manager could be useful.  





  with the debug mode.
------------------------ 

now i read your answer, i reinstalled the 139 and put the debug to file option and tried to update by wifi.
the update worked immediatly oO
so i was again in 140, tried to update to 139, when i say yes nothing happened so NON OK. same to 138 NON OK and 133 NON OK ... but i could update from 140 to 140 OK.
 
retry, 140 to 139 NON OK but 140 to 138 OK. 

so now the device run FW138, the firmware 140 could be downloaded despite i already have the  D0000140.FW file on the device, and at the "start update process" prompt the yes make the menu disappear (so the bug we are searching) and when i check the "install update" menu i have no more D0000140.FW file. 
that means the bug happen when we press the yes button not before nor when we press cancel button. 

update test again from FW138: connection error, reboot and miracle i could update from138 to 140 after validating "start update process".

here is the log (wifi pwd erased ) https://transfert.free.fr/download/GZKtochgADZRIipXQ2Oh0fnpX4

note: the battery was full yesterday evening and now it is 61%, i know battery optimisation are not a priority but there will be a issue soon.

in the debug mode, on each line you should put the number of the firmware like [140] or [139] to easy the reading and the debug.
and perhaps trace the menus and the options chosen.

**Comment by theosch1** [comment]
I also had the timeout problem so far, not when I downloaded Nr. 143. Maybe it was my near distance to the router that helped this .

**Comment by ukeconsulting** [comment]
with FW143, i tried many times to install the firmware 146 by wifi, when the download is complete nor yes or "no+install manually" work (no fw 146 shown on the fw list). 

on the pink, i could succeed after about 10 tests with yes.
on the orange one, i tried a new method : after download, i push reset. so i can see the fw146 in the manual update list, but i am unable to update: nothing happen. so i retry to download it by wifi and retry yes and... it worked : updating to 146 oO

**Comment by theosch1** [comment]
Hallo, the time-out problem probably is due to data error, which can not be managed. To my experience this problem can be minimized by positioning the strato close to the wifi source. This time the upload worked at the second try. Unforunately you have to restart before the next try, otherwise timeout is shown already when looking for a new version. This looks like a managable bug. I didn't try if reset works as well.

 
 

Gesendet: Montag, 30. August 2021 um 13:44 Uhr
Von: "Romain" ***@***.***>
An: "fhorinek/BB" ***@***.***>
Cc: "theosch1" ***@***.***>, "Comment" ***@***.***>
Betreff: Re: [fhorinek/BB] often unable to download firmware on wifi due to "connection time out" (#49)


 

with FW143, i tried many times to install the firmware 146 by wifi, when the download is complete nor yes or "no+install manually" work (no fw 146 shown on the fw list).

on the pink, i could succeed after about 10 tests with yes.
on the orange one, i tried a new method : after download, i push reset. so i can see the fw146 in the manual update list, but i am unable to update: nothing happen. so i retry to download it by wifi and retry yes and... it worked : updating to 146 oO

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by stratobugs** [comment]
Placing it directly beside the WIFI source works usually.

But only a few meters away (still very very strong and stable WIFI with all other devices; router shows data connetions of >100 Mbit) the connetion to STRATO interrups and sometims even the device needs a restart because of "BUS ERRORS".

**Comment by stratobugs** [comment]
Today even updating STARTO right beside the WiFi source did not work.
Sometimes connection time out right from the beginning, sometimes while downloading the update file. Sometimes it needs to be restarted because of BUS ERRORS:

![Uploading E69ADAE7-7F3A-482E-8A48-57F2DD521904.jpeg…]()

**Comment by ukeconsulting** [comment]
FW146 to FW148:

- on the orange strato: 2 time out+reset (reset is quicker and work better than power off, power off seems to lead to more time out than reset)  and download+updating ok

- on the pink one: 2 or 3 time out+reset, then download OK but yes lead to nothing, no FW148 in manual update list :-(  one or two time out, re-download OK, this time i push "no" to update, FW148 in manual update list :-) try manual update: nothing seems to append. retry manual update more than 10s after and i got this: 

![IMG_20210901_124212_083 1](https://user-images.githubusercontent.com/17803490/131659928-e873d1a3-ce57-4bc1-a1d5-ebb3fba4a11f.jpg)

then reset, retry manual update, updating to 148 ok. 



note: i tested downloading close the router to far of it, not real influence, it seems more stable from far.

**Comment by fhorinek** [comment]
I am currently working to improve the wifi reliability. 
Just the remark: manual update only show the downloaded firmware in your device.
It is ment as fail safe if the new firmware have some serious issue.

**Comment by ukeconsulting** [comment]
> Just the remark: manual update only show the downloaded firmware in your device.
> It is ment as fail safe if the new firmware have some serious issue.

i use it for upgrading since the upgrade directly from OTA doesn't work all the time. 
the wifi stability is a problem but something happen between the successfull download and the upgrade by itself: i see the newly downloaded firmware only if i reset the device after OTA download, or sometime after puting 'no' after download.

**Comment by theosch1** [comment]
The battarie is lasting more than 12 h now. However, when recharging, I don't reach 100% anymore, the maximum load decreases. I hope that there is a proper batterie charging management in strato.  Cheers, Theo

**Comment by ukeconsulting** [comment]
update from 154 to 166: update made successfully on both device at first try :-)  no bug seen.

**Comment by ukeconsulting** [comment]
i didn't have any issue with the download of firmware since some firmware, no problem to close it for me.
